aspnetcore/blazor/handle-errors/samples_snapshot/3.x/product-details.razor

```diff
@@ -0,0 +1,40 @@
+@page "/product-details/{ProductId:int}"
+@using Microsoft.Extensions.Logging
+@inject IProductRepository ProductRepository
+@inject ILogger<ProductDetails> Logger
+
+@if (details != null)
+{
+    <h1>@details.ProductName</h1>
+    <p>@details.Description</p>
+}
+else if (loadFailed)
+{
+    <h1>Sorry, we could not load this product due to an error.</h1>
+}
+else
+{
+    <h1>Loading...</h1>
+}
+
+@code {
+    private ProductDetails details;
+    private bool loadFailed;
+
+    [Parameter]
+    public int ProductId { get; set; }
+
+    protected override async Task OnParametersSetAsync()
+    {
+        try
+        {
+            loadFailed = false;
+            details = await ProductRepository.GetProductByIdAsync(ProductId);
+        }
+        catch (Exception ex)
+        {
+            loadFailed = true;
+            Logger.LogWarning(ex, "Failed to load product {PRODUCT_ID}", ProductId);
```

Comment:
Is `PRODUCT_ID` consistent with the standard naming convention for structured logging in our docs?

Maybe it is; I don't know. In the product sources we'd use a name like `ProductID` instead.

Reply:
Yes ... consistent. We uppercase template placeholders in other docs.

Reply:
*Ooops!* ... I researched this. We started with uppercase template placeholders (the framework code at the time did both), then we changed it in logging topics to go with Pascal case.

Hard to remember a two-minute convo from **_two years ago!_** :smile: lol

I'll revert this on the updates.